test/downloader.ts

```diff
@@ -0,0 +1,190 @@
+import * as tmp from 'tmp';
+import tape from 'tape';
+import nock from 'nock';
+import fs from 'fs';
+import path from 'path';
+import { keccak256 } from 'js-sha3';
+import { https } from 'follow-redirects';
+import downloader from '../downloader';
+
+const assets = path.resolve(__dirname, 'resources/assets');
+
+tape.onFinish(() => {
+  if (!nock.isDone()) {
+    throw Error('expected requests were not performed');
+  }
+});
+
+function hash (filePath: string): string {
+  return '0x' + keccak256(fs.readFileSync(filePath, { encoding: 'binary' }));
+}
+
+function generateTestFile (t: tape.Test, content: string): tmp.FileResult {
+  // As the `keep` option is set to true the removeCallback must be called by the caller
+  // to cleanup the files after the test.
+  const file = tmp.fileSync({ template: 'soljson-XXXXXX.js', keep: true });
+  try {
+    fs.writeFileSync(file.name, content);
+  } catch (err) {
+    t.fail('error writing test file');
```

Comment:
I think that would be nice to know why the file could not be written, i.e, was lack of permission, lack of memory or something else?

Reply:
Yes, I will add the `err` in the error message.